internal/controller/nonadminbackup_controller.go

```diff
@@ -89,22 +90,28 @@ func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	reconcileExit, reconcileRequeue, reconcileErr := r.InitNonAdminBackup(ctx, rLog, &nab)
 	if reconcileRequeue {
 		return ctrl.Result{Requeue: true, RequeueAfter: requeueTimeSeconds * time.Second}, reconcileErr
-	} else if reconcileExit || reconcileErr != nil {
-		return ctrl.Result{}, reconcileErr
+	} else if reconcileExit && reconcileErr != nil {
+		return ctrl.Result{}, reconcile.TerminalError(reconcileErr)
```

Comment:
instead of adding another else here, can `reconcile.TerminalError` be applied in functions? For example, inside `InitNonAdminBackup` function

Reply:
from what I read (did not test), I think it can be applied to `nil` as well, so would not change else structure here, just wrap error in `reconcile.TerminalError`

Reply:
I've tested this and adding nil won't work as this will always produce ERROR in reconcile, so it's a no-go.